ankihub/gui/decks_dialog.py

```diff
@@ -589,24 +590,69 @@ def _setup_box_note_types(
         self.note_types_label = QLabel("<b>📝 Note Types</b>")
         self.add_note_type_btn = QPushButton("Publish note type")
         qconnect(self.add_note_type_btn.clicked, self._on_add_note_type_btn_clicked)
+        self._update_add_note_type_btn_state()
         self.add_field_btn = QPushButton("Publish field")
         qconnect(self.add_field_btn.clicked, self._on_add_field_btn_clicked)
+        self._update_add_field_btn_state()
         self.update_templates_btn = QPushButton("Publish style/template updates")
-        self._update_templates_btn_state()
         qconnect(
             self.update_templates_btn.clicked, self._on_update_templates_btn_clicked
         )
+        self._update_templates_btn_state()
         box.addWidget(self.note_types_label)
         box.addWidget(self.add_note_type_btn)
         box.addWidget(self.add_field_btn)
         box.addWidget(self.update_templates_btn)
 
         return box
 
-    def _update_templates_btn_state(self):
-        self.update_templates_btn.setEnabled(
-            bool(note_types_with_template_changes_for_deck(self._selected_ah_did()))
+    def _update_note_type_btn_state(self, button: QPushButton, enabled: bool):
+        button.setEnabled(enabled)
+        if enabled:
+            button.setToolTip("")
+        else:
+            button.setToolTip("Nothing to update to AnkiHub")
+
+    def _get_note_type_names_for_add_note_type_btn(self) -> List[str]:
+        return self._get_note_type_names_for_deck(
+            self._selected_ah_did(), assigned_to_deck=False
+        )
+
+    def _update_add_note_type_btn_state(self):
+        enabled = bool(self._get_note_type_names_for_add_note_type_btn())
+        self._update_note_type_btn_state(self.add_note_type_btn, enabled)
+
+    def _get_note_type_names_for_add_field_type_btn(self) -> List[str]:
+        names_and_ids = self._get_note_type_names_and_ids_for_deck(
+            self._selected_ah_did(), assigned_to_deck=True
         )
+        filtered_names = []
+        for name_and_id in names_and_ids:
+            note_type = aqt.mw.col.models.get(NotetypeId(name_and_id.id))
+            if new_fields_for_note_type(self._selected_ah_did(), note_type):
+                filtered_names.append(name_and_id.name)
+
+        return filtered_names
+
+    def _update_add_field_btn_state(self):
+        enabled = bool(self._get_note_type_names_for_add_field_type_btn())
+        self._update_note_type_btn_state(self.add_field_btn, enabled)
+
+    def _get_note_type_names_for_update_templates_btn(self) -> List[str]:
+        mids_with_updates = note_types_with_template_changes_for_deck(
+            self._selected_ah_did()
+        )
+        return [
+            n.name
+            for n in self._get_note_type_names_and_ids_for_deck(
+                self._selected_ah_did(), assigned_to_deck=True
+            )
+            if n.id in mids_with_updates
+        ]
```

Comment:
🔧 This function is returning True after the update is complete, because the note_type has "\n\n" at the end. 

![image](https://github.com/user-attachments/assets/31d89691-94c0-4660-b25e-90d95e52d1da)

Because of it the state of the button is not updated.

https://github.com/user-attachments/assets/7bf8b391-b973-4ae4-9406-977cc8438a84

Reply:
This is probably just the result of the backend modifying `tmpls` and `css` (probably sanitization). `update_note_type_templates_and_styles()` assumes the backend does no further modifications of its own so we're not updating the local note type based on the one returned by `client.update_note_type()`.